pkg/pebinaryexecutor/pebinary.go

```diff
@@ -212,22 +223,110 @@ func (p *PEBinaryConfig) KubeProxy(ctx context.Context, args []string) error {
 
 	args = append(getArgs(extraArgs), args...)
 
-	logrus.Infof("Running RKE2 kube-proxy %s", args)
-	go func() {
+	win.ProcessWaitGroup.StartWithContext(ctx, func(ctx context.Context) {
 		outputFile := logging.GetLogger(filepath.Join(p.DataDir, "agent", "logs", "kube-proxy.log"), 50)
 		for {
+			logrus.Infof("Running RKE2 kube-proxy %s", args)
 			cmd := exec.CommandContext(ctx, filepath.Join("c:\\", p.DataDir, "bin", "kube-proxy.exe"), args...)
 			cmd.Stdout = outputFile
 			cmd.Stderr = outputFile
 			err := cmd.Run()
 			logrus.Errorf("kube-proxy exited: %v", err)
-			time.Sleep(5 * time.Second)
+
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				time.Sleep(5 * time.Second)
+			}
 		}
-	}()
+	})
 
 	return nil
 }
 
+// Docker starts cri-dockerd as implemented in the k3s cridockerd package
+func (p *PEBinaryConfig) Docker(ctx context.Context, cfg *config.Node) error {
+	return cridockerd.Run(ctx, cfg)
+}
+
+// Containerd configures and starts containerd.
+func (p *PEBinaryConfig) Containerd(ctx context.Context, cfg *config.Node) error {
+	args := getContainerdArgs(cfg)
+	stdOut := io.Writer(os.Stdout)
+	stdErr := io.Writer(os.Stderr)
```

Comment:
Did you manage to get containerd logs in Windows when using this? I had a lot of trouble and was unable to do it, so by default I decided to dump logs in files (kube-proxy, etc)

Reply:
Yes, I was able to find the logs in `C:\var\lib\rancher\rke2\agent\containerd\containerd.log`